### PR TITLE
[SIW-1201] add wallet test

### PR DESCRIPTION
**Pull request by PierpaoloPirasSp**
## WALLET TESTING

Tests have been added for the WALLET section.

## List of Implemented Tests

1. ### Accordion Component
   The accordion component has been reimplemented to track the opening and closing of the accordions themselves, and the relevant test has been added.
2. ### Wallet Revocation Error
   Tests for the displayed copy and the buttons present on the error page during wallet deactivation.
3. ### Wallet Thank You Page
   Tests for the displayed copy and the buttons present on the success page during wallet deactivation.
4. ### getWalletStatus
   Test for the getWalletStatus utility, which is used to track the status of the wallet in mix panel events.
5. ### ProfileCards
   Test for the ProfileCards component to verify that the wallet deactivation card is displayed only when the wallet is active.
6. ### HomeWalletCard
   Test for the HomeWalletCard component to verify that the card is displayed with the correct copy based on the wallet status.

### CODE REVIEW

**Comment by gunzip** [comment]
@PierpaoloPirasSp it looks like there are conflicts and we cannot merge this until they're solved

**Comment by PierpaoloPirasSp** [comment]
@gunzip Conflicts are resolved

File: openApi/openapi.yaml [hunk 1]
```diff
@@ -264,9 +264,11 @@ paths:
       operationId: setWalletInstanceStatus
       requestBody:
         content:
-          application/json:
-            schema:
-              $ref: '#/components/schemas/SetWalletInstanceStatusData'
+          text/plain:
```
Comment (gunzip) [inline openApi/openapi.yaml:267]: please revert to the previous version of this openapi spec (moreover, this is invalid synatx)

see https://github.com/pagopa/io-wallet/blob/master/apps/io-wallet-web-func/openapi.yaml#L45

File: openApi/openapi.yaml [hunk 2]
```diff
@@ -379,8 +381,8 @@ components:
         is_revoked:
           type: boolean
       required:
-        - id
-        - is_revoked
+          - id
```
Comment (gunzip) [inline openApi/openapi.yaml:384]: ditto, revert to previous working version

File: src/api/webProfileApiClient.ts [hunk 3]
```diff
@@ -151,7 +150,7 @@ export const WebProfileApi = {
     const result = await webProfileApiClient.getCurrentWalletInstanceStatus({});
     return extractResponse(result);
   },
-  setWalletInstanceStatus: async (id: SetWalletInstanceStatusDataEnum) => {
+  setWalletInstanceStatus: async (id: string) => {
```
Comment (gunzip) [inline src/api/webProfileApiClient.ts:153]: revert to previous version

File: src/app/[locale]/(pages)/revoca-wallet/errore/__test__/errorPage.test.tsx [hunk 4]
```diff
@@ -0,0 +1,19 @@
+import { renderWithProviders } from "@/app/[locale]/_utils/test-utils";
```
Comment (gunzip) [inline src/app/[locale]/(pages)/revoca-wallet/errore/__test__/errorPage.test.tsx:1]: use dashes for file names, ie. `error-page.test.tsx`

File: src/app/[locale]/(pages)/revoca-wallet/operazione-completata/__test__/thankYouPage.test.tsx [hunk 5]
```diff
@@ -0,0 +1,19 @@
+import { renderWithProviders } from "@/app/[locale]/_utils/test-utils";
+import { screen } from "@testing-library/react";
+
+import * as it from '../../../../../../dictionaries/it.json';
+import ThankYouPage from "../page";
+
+
+describe('Revoke wallet ThankYou page', () => {
+
+  test('should render error page', async () => {
```
Comment (gunzip) [inline src/app/[locale]/(pages)/revoca-wallet/operazione-completata/__test__/thankYouPage.test.tsx:10]:
```suggestion
  test('should render thank you page', async () => {
```

File: src/app/[locale]/_component/homeWalletCard/__test__/homeWalletCard.test.tsx [hunk 6]
```diff
@@ -0,0 +1,57 @@
+import React from 'react';
```
Comment (gunzip) [inline src/app/[locale]/_component/homeWalletCard/__test__/homeWalletCard.test.tsx:1]: `home-wallet-card.test.tsx`

File: src/app/[locale]/_component/profileCards/__test__/profileCards.test.tsx [hunk 7]
```diff
@@ -0,0 +1,38 @@
+import React from 'react';
```
Comment (gunzip) [inline src/app/[locale]/_component/profileCards/__test__/profileCards.test.tsx:1]: `profile-cards.test.tsx`

File: src/app/[locale]/_component/profileCards/__test__/profileCards.test.tsx [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+};
+
+
+describe('ProfileCards component', () => {
+
+  test('should render disable wallet card', async () => {
+    await renderWithProviders(<div data-testid="profile-cards"><ProfileCards {...activeWalletProps} /></div>);
+    const element = await screen.findByTestId('profile-cards')
+    expect(element).toBeInTheDocument();
+    const disbaleTitle = screen.getAllByText(it.itwallet.common.disablewallet)
```
Comment (gunzip) [inline src/app/[locale]/_component/profileCards/__test__/profileCards.test.tsx:24]:
```suggestion
    const disableTitle = screen.getAllByText(it.itwallet.common.disablewallet)
```

**Review by gunzip — CHANGES_REQUESTED** [review]

**Review by gunzip — APPROVED** [review]

**Pull request merged**